src/hooks/useHighlightedItemDetector.tsx

```diff
@@ -0,0 +1,26 @@
+import { useMapEditorContext } from "../MapEditorContext";
+import { CanvasRenderableItem } from "../models/CanvasRenderableItem";
+import { Nullable } from "../models/Nullable";
+
+export const useHighlightedItemDetector = (): Nullable<
+    CanvasRenderableItem
+> => {
+    const { padding, mousePosition, platforms } = useMapEditorContext();
+    const [mouseX, mouseY] = mousePosition;
+
+    const highlightedPlatform = platforms.find(({ x, y, width, height }) => {
+        const left = x - padding;
+        const top = y - padding;
+        const right = x + width + padding;
+        const bottom = y + height + padding;
+
+        return mouseX >= left && mouseX <= right && mouseY >= top
+            && mouseY <= bottom;
+    });
+
+    if (highlightedPlatform) {
+        return highlightedPlatform;
+    }
+
+    return null;
+};
```

Comment:
Is anything preventing this from running on every render of the consuming component for this hook? Would it be necessary to add a `useEffect` or am I missing something? Here's a suggestion such that the hook runs when the mouse position changes (which I'm not sure is better than every render, or maybe is even worse, I dunno how the canvas works)

```suggestion
export const useHighlightedItemDetector = (): Nullable<
    CanvasRenderableItem
> => {
    const { padding, mousePosition, platforms } = useMapEditorContext();
    const [mouseX, mouseY] = mousePosition;
    const [highlightedPlatform, setHighlightedPlatform] = useState<Nullable<CanvasRenderableItem>>(null);
    
    useEffect(() => {
        const highlightedPlatform = platforms.find(({ x, y, width, height }) => {
        const left = x - padding;
        const top = y - padding;
        const right = x + width + padding;
        const bottom = y + height + padding;

        return mouseX >= left && mouseX <= right && mouseY >= top
            && mouseY <= bottom;
        });
        
        setHighlightedPlatform(highlightedPlatform);
    
    }, [mousePosition]}    

    return highlightedPlatform;
};
```

Reply:
I think you might be right that it should be in some kind of `useEffect` hook but I think it would also need to include platforms just in case you add a platform under your cursor without moving it

Reply:
Good call